dask/dataframe/io/parquet/core.py

```diff
@@ -1042,21 +1032,26 @@ def get_engine(engine):
 
         if engine in ("pyarrow", "arrow"):
             engine = "pyarrow-dataset"
-        elif pa_version.major >= 5 and engine == "pyarrow-legacy":
-            warnings.warn(
-                "`ArrowLegacyEngine` ('pyarrow-legacy') is deprecated for "
-                "pyarrow>=5 and will be removed in a future release. Please "
-                "use `engine='pyarrow'` or `engine='pyarrow-dataset'`.",
-                FutureWarning,
-            )
 
         if engine == "pyarrow-dataset":
+            if pa_version.major < 1:
+                raise ImportError(
+                    f"pyarrow-{pa_version.major} does not suppot the "
+                    f"pyarrow.dataset API. Please install pyarrow>=1."
+                )
+
             from .arrow import ArrowDatasetEngine
 
             _ENGINES[engine] = eng = ArrowDatasetEngine
         else:
             from .arrow import ArrowLegacyEngine
 
+            warnings.warn(
+                "`ArrowLegacyEngine` ('pyarrow-legacy') is deprecated "
+                "and will be removed in the near future. Please use "
+                "`engine='pyarrow'` instead.",
+                FutureWarning,
```

Comment:
```suggestion
                FutureWarning,
                stacklevel=3,
```

(don't know if you do stacklevels in dask)

Reply:
I took this suggestion, but I also don't know :)